GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift

```diff
@@ -0,0 +1,23 @@
+import UIKit
+
+final class OnHoldOverlayVisualEffectView: UIVisualEffectView {
+    private var animator = UIViewPropertyAnimator(
+        duration: 1,
+        curve: .linear
+    )
+
+    override func draw(_ rect: CGRect) {
+        super.draw(rect)
+
+        effect = nil
+        animator.stopAnimation(true)
+        animator.addAnimations { [weak self] in
+            self?.effect = UIBlurEffect(style: .regular)
+        }
+        animator.fractionComplete = 0.1
+    }
+
+    deinit {
+        animator.stopAnimation(true)
```

Comment:
why do we need to stop animation explicitly on deinitialization? will it cause any issues if we will deinit it without stopping the animator?

Reply:
Yes, in case of incorrect memory usage where the view would stay alive (eg. retain cycle), this at least ensures that animation will be ended properly instead of running and wasting resources forever

Reply:
but in case of a memory leak `deinit` method will not execute or do I miss something?

Reply:
ah, you are correct. I guess I just saw online that it was common practice without really questioning it